lib/spack/spack/util/spack_yaml.py

```diff
@@ -204,6 +256,69 @@ def ignore_aliases(self, _data):
 OrderedLineDumper.add_representer(syaml_dict, OrderedLineDumper.represent_dict)
 OrderedLineDumper.add_representer(syaml_list, OrderedLineDumper.represent_list)
 OrderedLineDumper.add_representer(syaml_str, OrderedLineDumper.represent_str)
+OrderedLineDumper.add_representer(syaml_int, OrderedLineDumper.represent_int)
+
+
+def file_line(mark):
+    """Format a mark as <file>:<line> information."""
+    result = mark.name
+    if mark.line:
+        result += ':' + str(mark.line)
+    return result
+
+
+#: Global for interactions between FirstPassDumper and AnnotatedDumper
+#: This is nasty but YAML doesn't give us many ways to pass arguments.
+_annotations = []
```

Comment:
The only place I see `_annotations` being written is by `FirstPassDumper`, the only place I see it read is in `dump_annotations`, which also declares `FirstPassDumper`. So I'm not sure why this has to be global.